sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -113,38 +115,25 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    * @throws Error if an open receiver is already existing for given sessionId.
    */
   private constructor(
-    context: ClientEntityContext,
+    private _messageSession: MessageSession,
+    private _context: ClientEntityContext,
     public receiveMode: "peekLock" | "receiveAndDelete",
-    private _sessionOptions: CreateSessionReceiverOptions,
-    retryOptions: RetryOptions = {}
+    private _retryOptions: RetryOptions = {}
   ) {
-    throwErrorIfConnectionClosed(context.namespace);
-    this._context = context;
+    throwErrorIfConnectionClosed(this._context.namespace);
     this.entityPath = this._context.entityPath;
-    this._retryOptions = retryOptions;
-
-    if (this._sessionOptions.sessionId) {
-      this._sessionOptions.sessionId = String(this._sessionOptions.sessionId);
-
-      // Check if receiver for given session already exists
-      if (
-        this._context.messageSessions[this._sessionOptions.sessionId] &&
-        this._context.messageSessions[this._sessionOptions.sessionId].isOpen()
-      ) {
-        const errorMessage = getOpenSessionReceiverErrorMsg(
-          this._context.entityPath,
-          this._sessionOptions.sessionId
-        );
-        const error = new Error(errorMessage);
-        log.error(`[${this._context.namespace.connectionId}] %O`, error);
-        throw error;
-      }
+    // By this point, we should have a valid sessionId on the messageSession
+    // If not, the receiver cannot be used, so throw error.
+    if (_messageSession.sessionId == undefined) {
```

Comment:
As discussed, this is already checked in `MessageSession.init()`
I understand the need to ensure we are working with a valid sessionId by the time we get here.
So, I would recommend that we move the checks from inside `MessageSession.init()` to here.

Wait for #9766 to be merged before the move though.

Reply:
Validations inside init seem fine.
Moving the checks from `MessageSession.init()` to here would mean that we need to move the followup setup too which involve methods internal to MessageSession.
![image](https://user-images.githubusercontent.com/10452642/86411424-b7d90180-bc71-11ea-8733-b52ac4c6fe9a.png)
(and perhaps the log statements too.)

It would look something like this https://github.com/Azure/azure-sdk-for-js/pull/8849/commits/849930bd4d22265d70f12a31beed581e621311a6
Let me know if you don't like it, I'll revert the commit.

Reply:
One more option is to trust `_init` to do the proper validation related to the sessionIds, throw errors if required.
And not have any validations here at the constructor and just add comments saying init does the right validations.

Reply:
The follow ups could have stayed inside `init()` under an `if (receivedSessionId != undefined)` check right?

Reply:
Perhaps this?
`if (receivedSessionId != undefined && (this.sessionId === undefined || this.sessionId === receivedSessionId ))`

Reply:
Moved back to init with the new if check.